src/main/scala/com/gossiperl/client/GossiperlClient.scala

```diff
@@ -1,8 +1,36 @@
 package com.gossiperl.client
 
-trait GossiperlClient {
+import akka.actor._
+import akka.util.Timeout
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import concurrent.duration._
+import scala.concurrent.Promise
+import scala.util.{Failure, Success}
 
-  def withOverlay():Unit = {
+trait GossiperlClient extends LazyLogging {
+
+  val system = ActorSystem("gossiperl-system")
+  val supervisor = system.actorOf(Props[Supervisor], name=Supervisor.actorName)
+  import scala.concurrent.ExecutionContext.Implicits.global
+  implicit val timeout = Timeout(1 seconds)
+
+  def withOverlay( configuration: OverlayConfiguration ) ( action: Option[GossiperlProxy] => Unit ):Unit = {
+    system.actorSelection(s"/user/${Supervisor.actorName}").resolveOne( timeout.duration ) onComplete {
+      case Success(a) =>
+        val p = Promise[GossiperlProxy]
+        p.future onComplete {
+          case Success( proxy ) =>
+            action.apply( Some( proxy ) )
+          case Failure( ex ) =>
+            logger.error(s"Error while awaiting for gossiperl proxy with configuration $configuration.", ex)
+            action.apply( None )
+        }
+        a ! SupervisorProtocol.Connect( configuration, p )
```

Comment:
That was the idea initially. But how do I know which promise to complete without passing it around? Eventually, not sure if this can be done, something like:

    receive = {
      case ...
        a ! ...
        receive = {
        }
    }

But I don't think it is possible. I agree with the general argument. If there's a way to find which promise to complete, I'm happy to implement.